### ISSUE TITLE
Question about validation loss

### ISSUE THREAD

**Issue by liangdima** [Luchixiang/PCRL#9]
@Luchixiang  
 Hi,

Thanks for your impressive work.
I tired to reproduce the work and have some questions about the validation loss during pre-training stage.

I noticed the paper mentions "We employ the cosine annealing strategy for decreasing learning rate and stop the training when the validation loss does not change for 30 epochs. The checkpoints with lowest validation loss values are saved for finetuning." But it seems no related codes are included in the project. So there are some questions,
1. How did you build the valid dataset during pre-training stage.
2. Did you use any augmentations during validation of pre-trained model and finetune?
3. What dose the validation loss (for pre-training stage) consist of? Only the reconstruction loss of the ordinary encoder or the sum of the reconstruction loss of ordinary encoder, the momentum encoder and the mixup encoder?

Looking forword to your help!

**Comment by Luchixiang** [comment]
Sorry for my late reply and thanks for the interests. 
During the pertaining stage, we did use consine annealing strategy for decreasing learning rate and we pretrained for 240 epoches(when the training loss is converged). During the fine-tuning stage, the checkpoints with lowest validation loss values are saved for testing.

**Comment by liangdima** [comment]
> 
Thanks for your reply.
 It might be my fault that I misunderstood the description in the paper. So you mean that during pretraining stage, the model was pretrained by 240 epochs regularly, and validation is only took for finetune?

Besides, did you take any data augmentation during finetuning stage?

**Comment by Luchixiang** [comment]
Yes. During the fine-tune stage, for the different dataset, we use the different augmentation strategies. For example, we use flip, randresizedcrop rotation for chest14 dataset.

**Comment by liangdima** [comment]
Thanks for your reply.

I tried to reproduce the result of paper (chest14 --> chest14) but failed to achieve it.
I just rewrote the code for distributed training. But the intermediate visualisation results seems that the encoder-decoder network doesn't work well. ( I plotted the output of the decoder and the corresponding mask, which should be similar in ideal condition but not so well in reality.) I have double-checked my code but still not found what is the reason that  makes the results drop so much from the paper representation. If possible, would you like to send me a distributed version code  and the finetuning code? I would only use it for academic usage.

**Comment by Luchixiang** [comment]
Hi. Could you please tell me your email? I will send the fine-tune code (a little bit messy as we don't clean it)to you. I'm sorry that we don't have the distributed version code. You can send the email to 984335792@qq.com if you don't want to make your email public.

**Comment by liangdima** [comment]
Hi Lu, thanks for your kindly help. I have solved the problems and successfully reproduced the results of the paper. It's my fault that I got some misunderstandings of the finetuning stage before. Again, thanks for your remarkable works!

**Comment by Luchixiang** [comment]
Great
